src/python/WMCore/ReqMgr/Utils/Validation.py

```diff
@@ -186,3 +187,35 @@ def get_request_template_from_type(request_type, loc="WMSpec.StdSpecs"):
 
     result = create_json_template_spec(specArgs)
     return result
+
+def validateOutputDatasets(outDsets, dbsUrl):
+    """
+    Validate output datasets after all the other arguments have been
+    locally update during assignment
+    """
+    datatier = []
+    for dataset in outDsets:
+        tokens = dataset.split("/")
+        procds = tokens[2]
+        datatier.append(tokens[3])
+        try:
+            procdataset(procds)
+        except AssertionError as ex:
+            msg = "Bad output dataset name, check the processed dataset.\n %s" % str(ex)
+            raise InvalidSpecParameterValue(msg)
+
+    # Verify whether the output datatiers are available in DBS
+    _validateDatatier(datatier, dbsUrl)
+
+def _validateDatatier(datatier, dbsUrl):
+    """
+    _validateDatatier_
+
+    Provided a list of datatiers extracted from the outputDatasets, checks
+    whether they all exist in DBS already.
+    """
+    dbsReader = DBSReader(dbsUrl)
```

Comment:
Maybe we should use cache (memory cache) to save Tier name since there . We probably don't want to make this validation fail when there is network or DBS glitch